GitDepend/Visitors/CleanDependencyVisitor.cs

```diff
@@ -45,41 +45,52 @@ public CleanDependencyVisitor(bool dryRun, bool force, bool cleanFiles, bool cle
         }
 
         /// <summary>
-        /// The return code
-        /// </summary>
-        public ReturnCode ReturnCode { get; set; }
-
-        /// <summary>
-        /// Visits a project dependency.
+        /// Visists a project.
         /// </summary>
         /// <param name="directory">The directory of the project.</param>
-        /// <param name="dependency">The <see cref="Dependency" /> to visit.</param>
+        /// <param name="config">The <see cref="GitDependFile" /> with project configuration information.</param>
         /// <returns>
         /// The return code.
         /// </returns>
         /// <exception cref="NotImplementedException"></exception>
-        public ReturnCode VisitDependency(string directory, Dependency dependency)
+        public new ReturnCode VisitProject(string directory, GitDependFile config)
         {
+            _git.WorkingDirectory = directory;
+            if (_dependencyNameToClean.Count > 0)
+            {
+                foreach (var item in _dependencyNameToClean)
+                {
+                    if (item == config.Name)
+                    {
+                        return _git.Clean(_dryRun, _force, _cleanFiles, _cleanDirectory);
```

Comment:
can safely execute this command without checking the dependency names. You are guaranteed that the dependency name matches if you got here.